frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -315,6 +315,9 @@ func Dockerfile2LLB(ctx context.Context, dt []byte, opt ConvertOpt) (*llb.State,
 			k, v := parseKeyValue(env)
 			d.state = d.state.AddEnv(k, v)
 		}
+		if val, ok := opt.BuildArgs["HOSTNAME"]; ok {
```

Comment:
why upper chars?

Reply:
Should these "special" build-args be namespaces somehow? This could easily conflict with someone having a `ARG HOSTNAME` in their dockerfile (is that a concern?)

Reply:
This should be a frontend opt rather than build-arg?

Reply:
I choose upper chars because bash always uses HOSTNAME to specifies it. And it seems it is easily causing confliction? @AkihiroSuda @thaJeztah what's your suggestion, use lower chars or other special names?

frontend opt sounds better, I'll update

Reply:
@AkihiroSuda @thaJeztah I updated to `--opt contexthostname=`, is it better?

Reply:
Why "context" prefix?

Reply:
This hostname is only valid in this building context, so I added the "context" prefix. Is it OK?

Reply:
I think "just" `hostname` would be ok from a UX perspective, as long as we document it properly (if the option is not set, the default is `buildkitsandbox`, and the hostname is only set during build, and not persisted in the resulting image)

Reply:
@thaJeztah Thanks for your comments! In the newest patch I've updated it to `hostname`, now we can use it as `buildctl build --opt hostname=blabla`

Reply:
Please use something more specific. Eg. `exec-hostname` `run-hostname`